src/objective-c/tests/CoreTests/CoreTests/EventEngineTimerTests.mm

```diff
@@ -40,12 +47,87 @@ + (void)tearDown {
 }
 
 - (void)setUp {
+  signaled_ = false;
+  _engine = absl::make_unique<grpc_event_engine::experimental::LibuvEventEngine>();
 }
 
 - (void)tearDown {
+  _engine = nullptr;
+}
+
+- (void)testImmediateCallbackIsExecutedQuickly {
+  grpc_core::MutexLock lock(&mu_);
+  _engine->RunAt(absl::Now(), [self]() {
+    grpc_core::MutexLock lock(&mu_);
+    signaled_ = true;
+    cv_.Signal();
+  });
+  cv_.WaitWithTimeout(&mu_, absl::Seconds(5));
+  XCTAssertTrue(signaled_);
+}
+
+- (void)testSupportsCancellation {
+  auto handle = _engine->RunAt(absl::InfiniteFuture(), []() {});
+  XCTAssertTrue(_engine->Cancel(handle));
+}
+
+- (void)testCancelledCallbackIsNotExecuted {
+  {
+    auto handle = _engine->RunAt(absl::InfiniteFuture(), [self]() {
+      grpc_core::MutexLock lock(&mu_);
+      signaled_ = true;
+    });
+    XCTAssertTrue(_engine->Cancel(handle));
+    _engine = nullptr;
+  }
+  // The engine is deleted, and all closures should have been flushed
```

Comment:
nit: do you mind copy/moving this comment where you set the engine to nullptr so readers can understand the intent of setting it to null? (below too)

Reply:
+1